### DISCUSSION TITLE
[feat] Plugin initialization order

### DISCUSSION THREAD

**Discussion by richerfu** [tauri-apps/tauri#5420] [Q&A]
### Describe the problem

```rust
    tauri::Builder::default()
        .plugin(plugin::a::init())
        .plugin(plugin::b::init())
        .plugin(plugin::c::init())
        .plugin(plugin::d::init())
        .setup(|app| {
            Ok(())
        })
        .invoke_handler(tauri::generate_handler![cmd::greet])
        .run(context)
        .expect("error while running tauri application");
```

### Describe the solution you'd like

Just like this, i provide some plugins.I hope them can be called with `a -> b -> c -> d`

**Comment by JonasKruckenberg** [comment]
I don't quite understand what you're problem is, but assuming you're asking wether plugin `b` can access `a`'s state (and `c` `b`'s state and so on) then yes, they can. 
Given that the types are public (i.e. accessible to the other plugins) you can access all state from the previously registered plugins.

In your main `setup` closure you have access to *all* public state.

**Reply by FabianLars** [reply]
Here's the relevant discord thread: https://discord.com/channels/616186924390023171/1025354522589151242

**Reply by richerfu** [reply]
> Here's the relevant discord thread: https://discord.com/channels/616186924390023171/1025354522589151242

ok, i got it.Thanks!
